docs_src/design/adr/013-Device-Service-Events-Message-Bus.md

```diff
@@ -0,0 +1,215 @@
+
+The App Service SDK will be enhanced for the secure Message Bus connection described below. See **[Secure Connections](#secure-connections)** for details
+
+### Message Bus Topics
+
+> *Note: The change recommended here is not required for this design, but it provides a good opportunity to adopt it.*
+
+Currently Core Data publishes Events to the simple `events` topic. All Application Services running receive all Events published, whether they want them or not. The Events can be filtered out using the `FilterByDeviceName` pipeline function, but the Application Services still receive all the Events and process all the Events to some extent. This could cause load issues in a deployment with many devices and large volume of Events.
+
+Pub/Sub systems have advanced topic schema, which we can take advantage of to filter for just the Events the Application Service actual needs. If the publishers of Events add the `Device Name` to the topic in the form `edgex/events/<device-name>` then the Application Service can filter for just the Events from the device(s) it wants by only subscribing to those `Device Names`, i.e. `edgex/events/Random-Integer-Device` . If persistence is require, Core Data will subscribe using the `#` wild card, i.e. `edgex/events/#` , so that it receives all Events. 
+
+The Message Bus abstraction allows for multiple subscriptions, so an Application Service could specify to receive data from only specific devices by creating multiple subscriptions. i.e.  `edgex/Events/Random-Integer-Device` and  `edgex/Events/Random-Boolean-Device`. Currently the App SDK only allows for a single subscription topic to be configured, but that could easily be expanded to handle a list of subscriptions. See [Configuration](#configuration) section below for details. 
+
+Core Data's existing publishing of Events would also need to be changed to use this new topic schema. One challenge with this is Core Data doesn't currently know the `Device Name` when it receives a CBOR encoded event. This is because it doesn't decode the Event until after it publishes it to the Message Bus. The V2 API could be enhanced to require the `Device Name` in the HTTP header when content type is CBOR.
+
+### Configuration
+
+#### Device Services
+
+All Device services will have the following additional configuration to allow connecting and publishing to the Message Bus.
+
+##### [MessageQueue]
+
+A  MessageQueue section will be added, which is similar to that used in Core Data today. To enable secure connections, the `Username` & `Password` have been replaced with `Authmode` & `SecretPath`, See **[Secure Connections](#secure-connections)** section below for details. The added `Enabled` property controls whether the Device Service publishes to the Message Bus or POSTs to Core Data. 
+
+```toml
+[MessageQueue]
+Enabled = true
+Protocol = 'tcp'
```

Comment:
We might only need either `Protocol` or `Type` here.  The `Type` could decide the `Protocol`.

Reply:
Nope, with `type` of `MQTT` or `Redis Streams` you still have a choice of different protocols. But we might want to consider collapsing `Protocol`, `Host` & `Port` into `Url` as we did with the MQTT Trigger.